.github/workflows/test-linux-cpu.yml

```diff
@@ -13,58 +13,50 @@ env:
   CHANNEL: "nightly"
 
 jobs:
+  setup:
+    runs-on: linux.12xlarge
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
   tests:
-    name: "Unit-tests on Linux CPU"
-    runs-on: [self-hosted, linux.12xlarge]
-    container:
-      image: pytorch/conda-builder:cpu
     strategy:
       matrix:
         py_vers: ["3.7", "3.8", "3.9", "3.10"]
       fail-fast: false
+    uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    with:
+      runner: linux.12xlarge
+      script: |
+        # Mark Build Directory Safe
+        git config --global --add safe.directory /__w/vision/vision
 
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-      - name: Set Release CHANNEL (for release)
-        if: ${{ (github.event_name == 'pull_request' && startsWith(github.base_ref, 'release')) || startsWith(github.ref, 'refs/heads/release') }}
-        run: |
-          echo "CHANNEL=test" >> "$GITHUB_ENV"
-      - name: Setup Conda
-        shell: bash -l {0}
-        env:
-          ENV_NAME: conda-env-${{ github.run_id }}
-          PY_VERS: ${{ matrix.py_vers }}
-        run: |
-          git config --global --add safe.directory /__w/vision/vision
-          . ~/miniconda3/etc/profile.d/conda.sh
-          conda create -yp ${ENV_NAME} python=${PY_VERS} numpy libpng jpeg scipy
-          echo "CONDA_RUN=conda run -p ${ENV_NAME}" >> "$GITHUB_ENV"
-      - name: Install TorchVision
-        shell: bash -l {0}
-        env:
-          VERSION: cpu
-          CUDATOOLKIT: cpuonly
-        run: |
-          # Needed for JPEG library detection as setup.py detects conda presence
-          # by running `shutil.which('conda')`
-          export PATH=~/miniconda3/bin:$PATH
-          set -ex
-          ${CONDA_RUN} conda install \
-            --yes \
-            -c "pytorch-${CHANNEL}" \
-            -c nvidia "pytorch-${CHANNEL}"::pytorch[build="*${VERSION}*"] \
-            "${CUDATOOLKIT}"
-          ${CONDA_RUN} python3 setup.py develop
-          ${CONDA_RUN} python3 -m pip install pytest pytest-mock 'av<10'
-      - name: Run tests
-        shell: bash -l {0}
-        env:
-          ENV_NAME: conda-env-${{ github.run_id }}
-          PY_VERS: ${{ matrix.py_vers }}
-        run: |
-          . ~/miniconda3/etc/profile.d/conda.sh
-          set -ex
-          ${CONDA_RUN} python3 -m torch.utils.collect_env
-          ${CONDA_RUN} python3 -m pytest --junitxml=test-results/junit.xml -v --durations 20
-          conda env remove -p ${ENV_NAME}
+        # Set up Environment Variables
+        export ENV_NAME=conda-env-"${{ github.run_id }}"
+        export PYTHON_VERSION="${{ matrix.py_vers }}"
+        export VERSION="cpu"
+        export CUDATOOLKIT="cpuonly"
+
+        # Set CHANNEL
+        if [[${{ (github.event_name == 'pull_request' && startsWith(github.base_ref, 'release')) || startsWith(github.ref, 'refs/heads/release') }} ]]; then
```

Comment:
These should actually be available as regular environment variables so it might be better to use those instead, see: https://docs.github.com/en/actions/learn-github-actions/environment-variables